src/sinks/datadog/logs/sink.rs

```diff
@@ -134,9 +134,17 @@ impl crate::sinks::util::encoding::Encoder<Vec<Event>> for JsonEncoding {
     fn encode_input(&self, mut input: Vec<Event>, writer: &mut dyn io::Write) -> io::Result<usize> {
         for event in input.iter_mut() {
             let log = event.as_mut_log();
-            log.rename_key(self.log_schema.message_key(), event_path!("message"));
-            log.rename_key(self.log_schema.host_key(), event_path!("host"));
-            if let Some(Value::Timestamp(ts)) = log.remove(self.log_schema.timestamp_key()) {
+            if let Some(message_path) = log.message_path() {
+                log.rename_key(message_path.as_str(), event_path!("message"));
+            }
+            if let Some(host_path) = log.host_path() {
+                log.rename_key(host_path.as_str(), event_path!("host"));
+            }
```

Comment:
🤔 don't we sort of need this to be infallible, payloads without these would be invalid. Shouldn't we handle that here?

Reply:
The problem is that these can only return `None` in the case that semantic meaning isn't defined for these fields. "Sink Requirements" can enforce that semantic meaning is set correctly for sinks, but that feature isn't stable / is opt-in right now.

I guess I should just make these panic if it doesn't exist, to match the current behavior in this sink.

Reply:
Well, that's awkward. Apparently I had this whole reply written, but never submitted the "review" 😅